core-web/libs/edit-content/src/lib/fields/dot-edit-content-radio-field/dot-edit-content-radio-field.component.spec.ts

```diff
@@ -0,0 +1,233 @@
+import { Spectator, createComponentFactory } from '@ngneat/spectator';
+
+import { ControlContainer, FormGroupDirective } from '@angular/forms';
+
+import { RadioButton } from 'primeng/radiobutton';
+
+import { DotEditContentRadioFieldComponent } from './dot-edit-content-radio-field.component';
+
+import {
+    RADIO_FIELD_BOOLEAN_MOCK,
+    RADIO_FIELD_FLOAT_MOCK,
+    RADIO_FIELD_INTEGER_MOCK,
+    RADIO_FIELD_TEXT_MOCK,
+    createFormGroupDirectiveMock
+} from '../../utils/mocks';
+
+describe('DotEditContentRadioFieldComponent', () => {
+    let spectator: Spectator<DotEditContentRadioFieldComponent>;
+
+    const createComponent = createComponentFactory({
+        component: DotEditContentRadioFieldComponent,
+        componentViewProviders: [
+            { provide: ControlContainer, useValue: createFormGroupDirectiveMock() }
+        ],
+        providers: [FormGroupDirective],
+        detectChanges: false
+    });
+
+    beforeEach(() => {
+        spectator = createComponent();
+    });
+    it('should if have an object without value and label, the label must be the same as the value, and of the same type', () => {
+        const RADIO_FIELD_TEXT_MOCK_WITHOUT_VALUE_AND_LABEL = {
+            ...RADIO_FIELD_TEXT_MOCK,
+            values: 'Dot'
+        };
+        spectator.setInput('field', RADIO_FIELD_TEXT_MOCK_WITHOUT_VALUE_AND_LABEL);
+        spectator.detectComponentChanges();
+
+        const expectedList = [
+            {
+                label: 'Dot',
+                value: 'Dot'
+            }
+        ];
+
+        expectedList.forEach((option) => {
+            expect(
+                spectator
+                    .queryAll(RadioButton)
+                    .find((radioOption) => radioOption.value === option.value)
+            ).toBeTruthy();
+        });
+    });
+
+    it('should render radio options', () => {
+        spectator.setInput('field', RADIO_FIELD_TEXT_MOCK);
+        spectator.detectComponentChanges();
+
+        const radioOptions = spectator.queryAll('p-radiobutton');
+        expect(radioOptions.length).toBe(2);
+    });
+
+    it('should dont have any value if the form value or defaultValue is null', () => {
+        spectator.setInput('field', RADIO_FIELD_TEXT_MOCK);
+        spectator.component.formControl.setValue(null);
+        spectator.detectComponentChanges();
+
+        const inputChecked = spectator.queryAll('div.p-radiobutton-checked');
+        expect(inputChecked.length).toBe(0);
+    });
+
+    it('should render radio selected if the form have value ', () => {
+        spectator.setInput('field', RADIO_FIELD_TEXT_MOCK);
+        spectator.component.formControl.setValue('one');
+        spectator.detectComponentChanges();
+
+        expect(spectator.component.formControl.value).toEqual('one');
+
+        const inputChecked = spectator.queryAll('div.p-radiobutton-checked');
+        expect(inputChecked.length).toBe(1);
```

Comment:
PrimeNG is a library that changes the CSS classes and HTML markup in their updates frequently. 

Using a selector like `div.p-radiobutton-checked` for the test might breaks the tests in an update without us know it... we can check the `checked` property of the component instance.

```suggestion
        const inputChecked = spectator.queryAll(RadioButton).map(radio => radio.checked);
        expect(inputChecked.length).toBe(1);
```